content/src/debug.js

```diff
@@ -123,15 +134,17 @@ var debug = window.ide = {
       panel: embedded ? 'auto' : true
     };
   },
-  trace: function(event, data) {
+  trace: function(event, data) { 
     detectStuckProgram();
     // This receives events for the new debugger to use.
     currentDebugId += 1;
-    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", data: "", seeeval:false};
+    var record = {line: 0, eventIndex: null, startCoords: [], endCoords: [], method: "", 
+        data: "", seeeval:false};
     traceEvents.push(event);
     currentEventIndex = traceEvents.length - 1;
     record.eventIndex = currentEventIndex;
     var lineno = traceEvents[currentEventIndex].location.first_line;
+    setTimeout(function() {view.createSlider(traceEvents, arrows, view.paneid("left"), debugRecordsByLineNo, targetWindow)}, 1000);
```

Comment:
Also, it's a bit odd for the slider to be passed the traceEvents array and the targetWindow and all that stuff.

It would be better if the slider was dealt with in the abstract, i.e., you just give it things like the # of ticks and the current tick position or something, and then you keep the traceEvents array within debug.js.